### PR TITLE
Fix usage of deprecated libavcodec functions

### PR DESCRIPTION
**Pull request by bjsowa**
Fixes web_video_server for Jazzy and Rolling distributions. 

I also dropped backward compatibility with old ffmpeg libraries to make the code easier to maintain.

### CODE REVIEW

**Comment by bjsowa** [comment]
@sea-bass This seems to work on Humble and Ubuntu 22.04 but Segfaults on Rolling on Ubuntu 24.04 when trying to use vp8, vp9 or h264 compression (mjpeg streaming works). Backtrace shows the seg fault occurs in `avformat_write_header` function. I want to fix this issue before merging this PR. Any help would be apprieciated.

**Comment by bjsowa** [comment]
I think I've got this finally working. Checked on Humble, Iron, Jazzy and Rolling. I can't get H264 to work (it never worked for me) but all of the other stream types (mjpeg, png, ros_compressed, vp8, vp9) are working.

**Comment by bjsowa** [comment]
`av_lockmgr_register` has been deprecated and does nothing since ffmpeg 4.0 (Ubuntu 20.04 uses ffmpeg 4.2.7) so I think we can safely remove it.

**Review by EzraBrooks — COMMENTED** [review]
Thanks for this contribution! You got to it before I was able to get my project's Jazzy upgrade underway :)

I will largely defer to @sea-bass here because I am not as well-versed in C++ as he is, but the changes seem sensical to me.

File: src/libav_streamer.cpp
Comment (sea-bass) [inline src/libav_streamer.cpp:246]: Did you want to keep these debug logs before merging, or did you intend to remove them?

File: src/libav_streamer.cpp
Comment (sea-bass) [inline src/libav_streamer.cpp:256]: This magic number of 0.95 would be nice to have as an actual constant defined in an anonymous namespace at the top of this file.

**Review by sea-bass — APPROVED** [review]
I was going to ask to update any READMEs/documentation to mention that h264 is removed... but that would assume the existence of that 😆 

Changes LGTM, and only have minor optional suggestions. Thank you for this!

File: src/libav_streamer.cpp
Comment (bjsowa) [reply src/libav_streamer.cpp:246]: The debug logs were backported from #103 . I don't see any reason to remove them.

File: src/libav_streamer.cpp
Comment (bjsowa) [reply src/libav_streamer.cpp:256]: I think this is out of scope for this PR. Instead of making it a constant I might add URL-controlled parameter for this (e.g. `playback_speed`) in the future.

**Pull request merged**
